### PR TITLE
[REF-2764] dep update 0.5.0

### PR DESCRIPTION
**Pull request by masenf**
Bump gunicorn dep to 22.0.0

Relock dependencies for CI testing

Upgrade pip for docker test to resolve backtracking issue

Filed #3244 since i couldn't upgrade fastapi

### CODE REVIEW

**Review by picklelo — DISMISSED** [review]

**Comment by masenf** [comment]
hmm definitely hitting pre-commit issues with this change... looking into it

**Comment by masenf** [comment]
Ah okay, we weren't running the linters with pydantic v2 installed prior to this patch.

**Comment by masenf** [comment]
@picklelo i think this one is good to go now, after fixing up a few issues with pyright running on pydantic v2.

just waiting for the last few slow CI runs to finish

**Review by picklelo — APPROVED** [review]

**Pull request merged**
